### ISSUE TITLE
Is dockerswarm.rocks still maintained?

### ISSUE THREAD

**Issue by pnbecker** [tiangolo/dockerswarm.rocks#89]
dockerswarm.rocks is great. Thank you for creating it. It seems there hasn't been any commit for more than two years now. Furthermore tools suggested like swarmprom are unmaintained and archived. How is the future of dockerswarm.rocks?

**Comment by Olegt0rr** [comment]
swarmprom - archived
swarmpit - abandoned

**Comment by disko** [comment]
SWIRL (https://github.com/cuigh/swirl) looks pretty much like swarmpit and seems to be actively maintained.

**Comment by saulshanabrook** [comment]
Swirl seems to be not that maintained either at this point ;(

**Comment by mpz** [comment]
Portainer have App Template with Grafana + Prometheus now:

https://www.portainer.io/blog/monitoring-a-swarm-cluster-with-prometheus-and-grafana

**Comment by tiangolo** [comment]
Hello! Thanks for the patience. Unfortunately, it seems Kubernetes won. 🥲 

I wouldn't recommend using Docker Swarm Mode for new projects. https://dockerswarm.rocks/swarm-or-kubernetes/

**Comment by jherazob** [comment]
Well shit...

**Comment by z3cka** [comment]
> Well shit...

I couldn't agree more. 

I'm sure there are ways to do Let's Encrypt with Traefik, or whatever reverse proxy, in k8s but it can't be a simple as the setup here at dockerswarm.rocks ... anyone know of any simple guides, with minikube or something, that accomplish more or less what we do here?

**Comment by s4ke** [comment]
> Hello! Thanks for the patience. Unfortunately, it seems Kubernetes won. 🥲
> 
> I wouldn't recommend using Docker Swarm Mode for new projects. https://dockerswarm.rocks/swarm-or-kubernetes/

@tiangolo While it's your project and website, and I will not tell you what to do, it makes me sad that one of the main resources we link over at https://github.com/BretFisher/awesome-swarm#articles-and-sample-code now actively discourages the use of a perfectly fine orchestrator. I really believe that there is a place for Docker Swarm. Sure, the development activity over at moby has slowed down a bit, but it is being maintained.

Maybe reword it a bit that Kubernetes has won, but Docker Swarm is at least a way to learn about clustering and people should check if it is enough but should not expect too many new features?

Again, your project, your choice. :)

**Comment by s4ke** [comment]
@z3cka if you need to handle TLS in your Swarm, I'd suggest to look at using caddy with https://github.com/lucaslorentz/caddy-docker-proxy to replace traefik and use a plugins like https://github.com/yroc92/postgres-storage for TLS storage.

**Comment by z3cka** [comment]
@s4ke I still have multiple swarms in prod, I'll rock those until support for Docker Swarm is dropped, which I don't think will happen any time soon. Thanks for the recommendation of Caddy. I've messed with Caddy a bit years ago for grabbing certs from LE, but only for small side projects. It seems that Caddy has evolved into a quite mature ecosystem, I'll check it out again within a Docker context. 👍

**Comment by persunde** [comment]
@Olegt0rr 

I am trying to make an updated version of Swarmprom here that I called [Swarmprom-Improved or SWIM](https://github.com/persunde/swarmprom-improved) here that should work with up to date versions of all the original Swarmprom software stack: https://github.com/persunde/swarmprom-improved

It is still WIP, maybe a few details in the guide is missing, but it is mostly done.

I hope it can be added to this amazing website, since it is really helpful for people who dont want to run Kubernetes but need some type of easy to run and deploy multi machine setup.

**Comment by tiangolo** [comment]
Thanks for the tip @s4ke! I just added a note pointing to https://github.com/BretFisher/awesome-swarm in https://github.com/tiangolo/dockerswarm.rocks/pull/142

Thanks everyone for the interest! :cake:

**Comment by marianhlavac** [comment]
@tiangolo Thanks for your work on this website so far, and for keeping it running even though you shifted your opinions.

**Comment by melutovich** [comment]
https://www.mirantis.com/blog/swarm-is-here-to-stay-and-keeps-getting-better-in-security-and-ease-of-operations
dated March 2024
"Mirantis is committed to providing the best enterprise support for Swarm for the long term — at least for the next three years — and for as long as it makes sense for our customers."

**Comment by z3cka** [comment]
@ChristianUlbrich I love it! Good on you keeping this alive. It would be great to get a link from the old site to this newly maintained one. I could imagine someone new to this space getting confused about the state of Docker Swarm. 

Here's to rocking Docker Swarm in 2025 and beyond!
